Third release analyses examples

Miagarciaru opened the pull request:
The HWW, TTbar (semilep channel), TTbarDilep (dilep channel) and ZTauTau analyses are modified in order to use the third release samples. As the migration process to third release gets done, the others analyses will be included in future commits. Thus, only HWW, TTbar (semilep), TTbarDilep and ZTauTau analyses will be working properly after the MR, but if someone wants to run the other analyses with the second release, they could use the 10fb-legacy-tag which is the back-up.

zlmarshall commented on Analysis/ZTauTauAnalysis/main_ZTauTauAnalysis.C:
```diff
@@ -15,8 +15,9 @@ void main_ZTauTauAnalysis(int proof = 0, int option= 0)
-  TString path = "http://opendata.cern.ch/eos/opendata/atlas/OutreachDatasets/2020-01-22/1lep1tau/";
-
+  //TString path = "/eos/user/g/garciarm/ntuple-production-samples/ntuples-open-data/";
+  TString path = "/eos/user/e/egramsta/OpenData/FEB2025/1LMET30/";
```
I think this should instead be `/eos/opendata/atlas/rucio/user/egramsta/`

zlmarshall commented on Analysis/ZTauTauAnalysis/ZTauTauAnalysis.h:
```diff
@@ -286,152 +304,126 @@ void ZTauTauAnalysis::Init(TTree *tree)
+  fChain->SetBranchAddress("jet_btag_quantile", &jet_btag_quantile, &b_jet_btag_quantile);
+  fChain->SetBranchAddress("jet_jvt", &jet_jvt, &b_jet_jvt);
+  /*
+  fChain->SetBranchAddress("largeRJet_n", &largeRJet_n, &b_largeRJet_n);
```
Low priority, 'if you're bored', kinda comment: if we don't think these sorts of things will be added back in at some point, I'd favor just deleting them instead of commenting them out (same with all the PROOF stuff)

zlmarshall commented on Analysis/TTbarDilepAnalysis/run.sh:
```diff
@@ -0,0 +1,49 @@
+        echo 'WARNING! After running, you need to add the samples using: hadd data.root dataA.root dataB.root dataC.root dataD.root'
+fi
+
+echo Should I use PROOF? \(will make things faster\)
```
It looks like ~all the PROOF stuff has been removed, so should this part of the run script also go away?

zlmarshall commented on Analysis/TTbarDilepAnalysis/main_TTbarDilepAnalysis.C:
```diff
@@ -0,0 +1,383 @@
+  /* Local path example */
+  //TString path = "/eos/user/g/garciarm/ntuple-production-samples/ntuples-open-data/";
+  //TString path = "/eos/user/e/egramsta/OpenData/2J2LMET30/";
+  TString path = "/eos/user/e/egramsta/OpenData/FEB2025/1LMET30/";
```
I think this should be `/eos/opendata/atlas/rucio/user/egramsta`, and I would favor removing all the old paths rather than leaving them there commented out.

zlmarshall commented on Analysis/TTbarAnalysis/main_TTbarAnalysis.C:
```diff
@@ -15,8 +15,9 @@ void main_TTbarAnalysis(int proof = 0, int option= 0)
-  TString path = "http://opendata.cern.ch/eos/opendata/atlas/OutreachDatasets/2020-01-22/1lep/";
-
+  //TString path = "http://opendata.cern.ch/eos/opendata/atlas/OutreachDatasets/2020-01-22/1lep/";
+  TString path = "/eos/user/e/egramsta/OpenData/FEB2025/1LMET30/";
```
As elsewhere, I think this should be `/eos/opendata/atlas/rucio/user/egramsta`

zlmarshall commented on Analysis/HWWAnalysis/main_HWWAnalysis.C:
```diff
@@ -3,19 +3,22 @@
+  //TString path = "http://opendata.cern.ch/eos/opendata/atlas/OutreachDatasets/2020-01-22/2lep/";
+  //TString path = "/eos/user/g/garciarm/ntuple-production-samples/ntuples-open-data/";
+  //TString path = "/eos/user/e/egramsta/OpenData/2J2LMET30/";
+  TString path = "/eos/user/e/egramsta/OpenData/FEB2025/1LMET30/";
```
As elsewhere, I think this should be `/eos/opendata/atlas/rucio/user/egramsta`, and we can tidy up the other paths instead of commenting them out

zlmarshall commented on Analysis/TTbarAnalysis/TTbarAnalysis.C:
```diff
@@ -48,294 +69,339 @@ Bool_t TTbarAnalysis::Process(Long64_t entry)
+		
+		int goodbjet1_index = goodbjet_index[0]; // leading b-jet
+		
+		// MTW > 30 GeV
```
It looks like this systematics block is all commented out. Is that right?

If so, maybe we should just remove it? We could also document how to use it and provide a switch to turn it on and off? I'm not sure what's best, but just having it in there and commented out isn't my favorite option.

Miagarciaru replied:
Hi @zlmarshall , I think we can remove it for the moment to get these analyses done for the coming set of analyses examples release and later we can discuss how we could implement it properly and include it in the following weeks.

Miagarciaru commented:
Hi @zlmarshall , thanks for your comments. I cleaned up some commented code that is not needed for the ZTauTau, HWW, TTBar and TTBarDilep analysis, so these examples can be read easier now. Also, I have modified the paths where the samples are going to be read and at last, I have removed the -1 or 1 mcWeight set up that was done manually for the second release. 
I think we could add documentation in some parts of the codes just for people who want to see how the analyses work in detail (see how to add variables, define and fill histograms, add cuts, etc), although there are a few comments that explain a little bit about some important parts. Anyway, we can complement the description in futures PRs, just in case that these analyses have to be ready soon, but for the moment I would say that it is ready now to be merged. Let me know what you think. Thanks!

zlmarshall reviewed (approved):
Thanks @Miagarciaru ! This is an impressive amount of work. I don't see anything in here that's still screaming for a change. I agree with you that it would be nice to put a load of comments throughout the code to explain what it's doing, but that is a _ton_ of work, and we should get this in without requiring that addition.

Just one last questions for me (though I'm going to hit 'approve' here): am I guessing right that you've tried out all these analyses and they work ok? 

A note for @Soap2G in case he's excited and bored, that CI/CD in this repo would be handy to have...

Thanks again, and nice work!
Zach

Miagarciaru commented:
Hi @zlmarshall, I ran the 4 analyses recently and noticed that each one is taking a lot of time (more or less 16 hours), but it is because the path you shared with me contains the ntuples without any skim, so the analyses are running over all the entries for each sample in this PR version. However, the validation plots are quite similar to the ones I got with the 1LMET30 skimmed samples. Also, I saw that there are a few histograms that were defined for systematics in the TTbarSemilep and ZTauTau analyses were also included in the Plotting tool, and it was producing empty histograms due to the fact that the systematics were not included in this version of the code (it would be for future implementations), so I have removed them from the list of histograms now. Thanks!

zlmarshall commented on Analysis/ZTauTauAnalysis/ZTauTauAnalysis.C:
```diff
@@ -53,229 +57,232 @@ Bool_t ZTauTauAnalysis::Process(Long64_t entry)
-		      }
-		    
-		    
-		    // systematic variations on tau pT
```
Ah, ok, here's where the systematic was being done. It's throwing random numbers based on the tau pT and systematic variation of the tau pT (we of course have the former, but not the latter) and picking out a "varied pT" based on that.  So for this to be working again, we will want a `tau_pt_syst` branch with a systematic variation on tau pT.

zlmarshall reviewed (approved):
Ok, I think we're there! Let's go! Thanks again @Miagarciaru
